tests/NCronJob.Tests/IntegrationTests.cs

```diff
@@ -784,30 +774,110 @@ await WaitForNthOrchestrationState(
+    public void SupportsValidRegistrations(Action<NCronJobOptionBuilder> register)
     {
-        ServiceCollection.AddNCronJob(n => n.AddJob<DummyJob>(p => p.WithCronExpression(Cron.AtEveryMinute)));
+        var act = () => ServiceCollection.AddNCronJob(register);
 
-        var runtimeRegistry = ServiceProvider.GetRequiredService<IRuntimeJobRegistry>();
+        act.ShouldNotThrow();
+    }
 
-        var successful = runtimeRegistry.TryRegister(n => n.AddJob<DummyJob>(p => p.WithCronExpression(Cron.AtEveryMinute)), out var exception);
+    public static TheoryData<Action<NCronJobOptionBuilder>> InvalidRegistrations = new()
+    {
+        {
+            // Names should be unique
+            s => s.AddJob<DummyJob>(p => p
+                    .WithName("JobName")
+                    .And
+                    .WithName("JobName"))
+        },
+        {
+            // Names should be unique
+            n => n.AddJob(() => { }, Cron.AtEveryMinute, jobName: "JobName")
+                  .AddJob(() => { }, Cron.AtMinute0, jobName: "JobName")
+        },
+        {
+            // Pure duplicate registration
+            s =>
+            {
+                s.AddJob<DummyJob>();
+                s.AddJob<DummyJob>();
+            }
+        },
+        {
+            // Pure duplicate registration
+            s => s.AddJob<DummyJob>(p => p
+                    .WithCronExpression(Cron.AtEveryMinute)
+                    .And
+                    .WithCronExpression(Cron.AtEveryMinute))
+        },
+        {
+            // No way to invoke DummyJob inambiguously
+            s => s.AddJob<DummyJob>(p => p
+                    .WithParameter("one")
+                    .And
+                    .WithParameter("two"))
+        },
+        {
+            // Pure duplicate registration
+            s => s.AddJob<DummyJob>(p => p
+                    .WithParameter("one")
+                    .And
+                    .WithParameter("one")).RunAtStartup()
+        },
+    };
 
-        successful.ShouldBeFalse();
-        exception.ShouldNotBeNull();
-    }
+    public static TheoryData<Action<NCronJobOptionBuilder>> ValidRegistrations = new()
+    {
+        {
+            s => s.AddJob<DummyJob>(p => p
+                    .WithCronExpression(Cron.AtEveryJanuaryTheFirst, TimeZoneInfo.FindSystemTimeZoneById("Eastern Standard Time"))
+                    .And
+                    .WithCronExpression(Cron.AtEveryJanuaryTheFirst, TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time")))
+        },
+        {
+            s => s.AddJob<DummyJob>(p => p
+                    .WithCronExpression(Cron.AtEveryMinute).WithParameter("one")
+                    .And
+                    .WithCronExpression(Cron.AtEveryMinute).WithParameter("two"))
+        },
+        {
+            s => s.AddJob<DummyJob>(p => p
+                    .WithParameter("one"))
+        },
+        // TODO: Re-implement this test in a UseNCronJob context
```

Comment:
@linkdotnet There *might* be a way to work around this, but it would trigger a some rewrite of the fluent interface. Basically, not storing the jobdefinition During the AddJob() call, but collect them, have RunAtStartup() execute against those and then add them to the jobregistry.

Not impossible, but not trivial either.

I'd rather postpone until V5. Besides, with #219 we'll have to revisit this whole fluent thingie

Thoughts?

Reply:
> Not impossible, but not trivial either.

Done with #255